src/modules/api/user/link/link.controller.ts

```diff
@@ -1,15 +1,45 @@
-import { IResult } from './../../../../common/interfaces/response';
-import { Controller, Post, Body, Param } from '@nestjs/common';
 import { LinkService } from './link.service';
+import { IResult } from './../../../../common/interfaces/response';
+import { Controller, Get, Post, Body, Param } from '@nestjs/common';
 import { LinkDTO } from './link.dto';
 
 @Controller('users/:userId/links')
 export class LinkController {
   constructor(private readonly linkService: LinkService) {}
+  @Get()
+  async getAllLinksForUserId(
+    @Param('userId') userId: string,
+  ): Promise<IResult<LinkDTO>> {
+    try {
+      const links = await this.linkService.getAllLinksOfUser(userId);
+      if (links) {
+        const linkArray = [];
+        links.forEach(link => {
+          const resp: LinkDTO = {
+            name: link.name,
+            customUrl: link.customUrl,
+            redirectsTo: link.redirectsTo,
+            status: link.status,
+            inPool: link.inPool,
+          };
+          linkArray.push(resp);
+        });
```

Comment:
How about using Map function instead of looping with for each and pushing them

Reply:
Okay i will check that

Reply:
```
 const linkArray = links.map(link => {
          const resp: LinkDTO = {
            name: link.name,
            customUrl: link.customUrl,
            redirectsTo: link.redirectsTo,
            status: link.status,
            inPool: link.inPool,
          };
          return resp;
        });
```

can be done like this I guess. 

```
 const linkArray = links.map(link =>{
            name: link.name,
            customUrl: link.customUrl,
            redirectsTo: link.redirectsTo,
            status: link.status,
            inPool: link.inPool,
        } as LinkDTO);
```

This is a bit cleaner, What do you think?

Reply:
yeah

Reply:
Just to make it cleaner. I'm not sure if it would effect the memory tho, because we are declaring a variable in the other way.